www/js/app.js

```diff
@@ -841,20 +839,26 @@ define(['jquery', 'zimArchiveLoader', 'util', 'uiUtil', 'cookies','abstractFiles
         var urlPath = regexpPath.test(dirEntry.url) ? urlPath = dirEntry.url.match(regexpPath)[1] : "";
         var baseUrl = dirEntry.namespace + "/" + urlPath;
 
-        //Inject base tag into html
+        // Inject base tag into html
         htmlArticle = htmlArticle.replace(/(<head[^>]*>\s*)/i, '$1<base href="' + baseUrl + '" />\r\n');
 
         // Display the article inside the web page.
-        document.getElementById("articleContent").contentDocument.documentElement.innerHTML = htmlArticle;
+        var iframeArticleContent = document.getElementById("articleContent");
+        var articleContent = iframeArticleContent.contentDocument;
+        articleContent.open();
+        articleContent.write(htmlArticle);
+        articleContent.close();
         
         // If the ServiceWorker is not useable, we need to fallback to parse the DOM
-        // to inject math images, and replace some links with javascript calls
+        // to inject images, CSS etc, and replace links with javascript calls
         if (contentInjectionMode === 'jquery') {
-            parseAnchorsJQuery();
-            loadImagesJQuery();
-            loadCSSJQuery();
-            //JavaScript loading currently disabled
-            //loadJavaScriptJQuery();            
+            iframeArticleContent.onload = function() {
```

Comment:
Hi @mossroy , thanks for your work on this! Can I just ask if you introduced this .onload function with the iframe in order to cure a race condition you observed with `articleContent.write()`? I've got absolutely no problem with doing it this way, but I'm curious as to whether you were still noticing the race problem with this way of injecting to the iframe -- I thought it was a problem pertaining to the loading of dummyArticle only, and went away once we got rid of dummyArticle. Again, no problem here, it looks like a good way of avoiding potential problems.

I'll try to do some testing of the PR this evening and will get back to you.

Reply:
In my first test, the links were simply not parsed (on my desktop Firefox 59). So they were leading to an error page, because the href was not found.
On the other hand, I tried to remove the onload trick, and can not reproduce the issue now...

So I can't say I'm 100% sure. I should test more

Reply:
I just reproduced it a few times, on several ZIM files

Reply:
OK, thanks for confirming. So it turns out document.write() is asynchronous, at least in the latest Firefox. Good to know.

Reply:
I reproduced that also on Firefox ESR 52.7.3 and Chrome 68, with the Ray Charles archive.
I suppose the browser enfines interpret the iframe in a separate thread, which makes sense.

I'm wondering if it would be necessary to move the statement "articleContent.close();" after the onload declarations. If the iframe loading is really in another thread, it might finish before we put the onload.
It's probably safer to move this line.

Reply:
I think I read somewhere that document.close(); may be necessary to trigger the document.onload event. May be wrong. But if it works as is, and #347 is not reproducible with this PR, then maybe we're OK...

Reply:
I think you're right about document.close() : the onload event is certainly fired after the document.close(). To illustrate my proposal, I've commited it in the branch : I've moved the document.close() after setting the onload